Enable set RPM rate limiter (#258)
Future-House/ldp: packages/lmi/README.md

### Rate limiting

Rate limiting helps regulate the usage of resources to various services and LLMs. The rate limiter supports both in-memory and Redis-based storage for cross-process rate limiting.

> Currently, `lmi` take into account the tokens used (the Token per Minute(TPM)) and the request handled (the Request per Minute (RPM)).

Reviewer: ```suggestion
Currently, `lmi` take into account the tokens used (Tokens per Minute (TPM)) and the requests handled (Requests per Minute (RPM)).
```

config = {
    "rate_limit": {
        "gpt-4": "100/minute",  # 100 tokens per minute
    },
    "request_limit": {
        "gpt-4": "5/minute",  # 5 request per minute
    },
}

llm = LiteLLMModel(name="gpt-4", config=config)
```

With `rate_limit` and `request_limit`, you can control both token consumption and request volume. You can also configure only one of them if needed.

2. Through the global rate limiter configuration:

```python
from lmi.rate_limiter import GLOBAL_LIMITER

GLOBAL_LIMITER.rate_config[("client", "gpt-4")] = "100/minute"  # 100 tokens per minute
GLOBAL_LIMITER.rate_config[("client|request", "gpt-4")] = (

>     "5/minute"  # 5 request per minute

Reviewer: ```suggestion
    "5/minute"  # 5 requests per minute
```

)
```

> With `client` and `client|request`, you can control both token consumption and request volume. You can also configure only one of them if needed.

Reviewer: ```suggestion
With `client` we rate limit only token consumption,
and with `client|request` we rate limit both token consumption and request volume.
You can also configure only one of them if needed.
```

config = {
    "rate_limit": {
        "gpt-4": "100/minute",  # 100 tokens per minute
    },
    "request_limit": {

>         "gpt-4": "5/minute",  # 5 request per minute

Reviewer 2: ```suggestion
        "gpt-4": "5/minute",  # 5 requests per minute
```

config = {
    "rate_limit": {
        "gpt-4": "100/minute",  # 100 tokens per minute
    },
    "request_limit": {

>         "gpt-4": "5/minute",  # 5 request per minute

Reviewer: ```suggestion
        "gpt-4": "5/minute",  # 5 requests per minute
```

# Example output:
{
    ("client|request", "gpt-4"): {
        "period_start": 1234567890,

>         "n_items_in_period": 1,

Reviewer: Can you explain in a comment what this `n_items_in_period` means for `client|request`?

Does it mean one request, one token, or both?

Author: > > Can we change the namespace for this to be `client|request` -- as request I think is a bit too general, and this refers to request from a particular client.
> 
> As suggested by @mskarlin, I have changed the namespace for `request_limit` from `"request"` to `"client|request"`. The `"client|request"` namespace is a single string and does not imply an "or" relationship in Python. It specifically refers to rate limiting _requests_. If this causes any confusion, I can adjust the naming to make it clearer.

I made an explanation here. Maybe it's better to change the namespace for `request_limit` and `rate_limit` to reveal their actual function. Do you have any suggestions?
